go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -611,138 +491,15 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
-				return nil, nil, nil, false, fmt.Errorf("failed to get pod addresses for pod %s on node: %s, err: %v",
-					podDesc, switchName, err)
-			}
-		}
-		needsNewMacOrIPAllocation := false
-
-		// ensure we have reserved the IPs found in OVN
-		if len(podIfAddrs) == 0 {
-			needsNewMacOrIPAllocation = true
-		} else if bnc.doesNetworkRequireIPAM() {
-			if err = bnc.lsManager.AllocateIPs(switchName, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
-				klog.Warningf("Unable to allocate IPs %s found on existing OVN port: %s, for pod %s on switch: %s"+
-					" error: %v", util.JoinIPNetIPs(podIfAddrs, " "), portName, podDesc, switchName, err)
-
-				needsNewMacOrIPAllocation = true
-			}
-		}
-		if needsNewMacOrIPAllocation {
-			if network != nil && network.IPRequest != nil && !bnc.doesNetworkRequireIPAM() {
-				klog.V(5).Infof("Will use static IP addresses for pod %s on a flatL2 topology without subnet defined", podDesc)
-				podIfAddrs, err = calculateStaticIPs(podDesc, network.IPRequest)
-				if err != nil {
-					return nil, nil, nil, false, err
-				}
-				podMac = util.IPAddrToHWAddr(podIfAddrs[0].IP)
-			} else {
-				// Previous attempts to use already configured IPs failed, need to assign new
-				generatedPodMac, generatedPodIfAddrs, err := bnc.assignPodAddresses(switchName)
-				if err != nil {
-					return nil, nil, nil, false, fmt.Errorf("failed to assign pod addresses for pod %s on switch: %s, err: %v",
-						podDesc, switchName, err)
-				}
-				if podMac == nil {
-					podMac = generatedPodMac
-				}
-				if len(generatedPodIfAddrs) > 0 {
-					podIfAddrs = generatedPodIfAddrs
-				}
-			}
-		}
-
-		releaseIPs = true
-		// handle error cases separately first to ensure binding to err, otherwise the
-		// defer will fail
-		if network != nil && network.MacRequest != "" {
-			podMac, err = calculateStaticMAC(podDesc, network.MacRequest)
-			if err != nil {
-				return nil, nil, nil, false, err
-			}
-		}
-		podAnnotation = &util.PodAnnotation{
-			IPs: podIfAddrs,
-			MAC: podMac,
-		}
-		var nodeSubnets []*net.IPNet
-		if nodeSubnets = bnc.lsManager.GetSwitchSubnets(switchName); nodeSubnets == nil && bnc.doesNetworkRequireIPAM() {
-			return nil, nil, nil, false, fmt.Errorf("cannot retrieve subnet for assigning gateway routes for pod %s, switch: %s",
-				podDesc, switchName)
-		}
-		err = bnc.addRoutesGatewayIP(pod, network, podAnnotation, nodeSubnets)
-		if err != nil {
-			return nil, nil, nil, false, err
-		}
-
-		klog.V(5).Infof("Annotation values: ip=%v ; mac=%s ; gw=%s",
-			podIfAddrs, podMac, podAnnotation.Gateways)
-		annoStart := time.Now()
-		err = bnc.updatePodAnnotationWithRetry(pod, podAnnotation, nadName)
-		podAnnoTime := time.Since(annoStart)
-		klog.Infof("[%s] addLogicalPort annotation time took %v", podDesc, podAnnoTime)
-		if err != nil {
-			return nil, nil, nil, false, err
-		}
-		releaseIPs = false
+	podAnnotation, annotationUpdated, err := bnc.allocateIPs(pod, existingLSP, nadName, network)
```

Comment:
I feel like you are replacing more than just "allocateIPs" as that logic also configures routes for the pod, mac, gateway ips. Maybe it should be allocateNetworkingForPod or something like that?